phpBB/phpbb/user.php

```diff
@@ -588,17 +588,42 @@ function add_lang_ext($ext_name, $lang_set, $use_db = false, $use_help = false)
 	*/
 	function format_date($gmepoch, $format = false, $forcedate = false)
 	{
+		global $phpbb_dispatcher;
 		static $utc;
 
 		if (!isset($utc))
 		{
 			$utc = new \DateTimeZone('UTC');
 		}
 
-		$time = new $this->datetime($this, '@' . (int) $gmepoch, $utc);
-		$time->setTimezone($this->timezone);
+		$format_date_override = false;
+		$function_arguments = func_get_args();
```

Comment:
Yes, I like it more as well, but my intention is to introduce a new parameter to the method, so there is no other way (one is unable to get arguments of the caller method).